### ISSUE TITLE
When a user creates an Appeal, no entry is made in the Instance Audit Log

### ISSUE THREAD

**Issue by ThisIsMissEm** [mastodon/mastodon#31461]
### Steps to reproduce the problem

1. Create an account warning for an account in response to a report.
2. From that warned account, appeal the account warning (Moderation strike)
3. Back as the Moderator, observe that you do not see in the report any update, even though the Appeal has been filed for the Account Warning.
...

### Expected behaviour

I should see an Audit Log entry saying "$username has appealed the [account warning]"

### Actual behaviour

No audit log entry occurs

### Detailed description

List of Appeals (as moderator) 

<img width="1215" alt="Screenshot 2024-08-17 at 02 36 06" src="https://github.com/user-attachments/assets/2c69316c-d67b-46fa-b2f7-3882558ee667">

Viewing the appeal (see #31460):
<img width="1261" alt="Screenshot 2024-08-17 at 02 36 31" src="https://github.com/user-attachments/assets/6dc0b397-4ed6-4be4-b440-26ea5954da8d">

Viewing the Report Audit Log associated with the Account Warning & Appeal:

<img width="868" alt="Screenshot 2024-08-17 at 02 43 09" src="https://github.com/user-attachments/assets/9b5e6397-f772-46dd-a1d0-d0129765c133">

No where else in the Report view do we see any mention of an open Appeal for the Account Warning sent in relation to the  Report.

### Mastodon instance

n/a

### Mastodon version

main

### Technical details

If this is happening on your own Mastodon server, please fill out those:
- Ruby version: 3.3.4
- Node.js version: 20-something

**Comment by ThisIsMissEm** [comment]
I would expect the Audit Log to show something like this:
<img width="849" alt="Screenshot 2024-08-17 at 02 45 40" src="https://github.com/user-attachments/assets/23c7f797-5534-4d81-b58c-ff6132251298">

(I'm not sure if it should be at the top or the bottom of the audit log, I can't remember how it's sorted, the key point is it should be in there)

**Comment by ThisIsMissEm** [comment]
Fixing this would also likely resolve the other half of https://github.com/mastodon/mastodon/issues/18761

**Comment by quintessence** [comment]
Also: beyond the Appeal itself being in the Audit Log, the outcome (Accept, Reject as well as if the account was Unlimited / Unfrozen / Unsuspended / etc.) should be in the Audit Log as well.

**Comment by ClearlyClaire** [comment]
That would make sense, but I can think of an issue with doing that.

Indeed, so far, we've always made audit logs to be about actions taken by moderators, and we list people who authored any audit log entry in `/admin/action_logs`.

Creating audit logs for people who appeal a moderation decision would change that substantially and would make the dropdown in `/admin/action_logs` huge and close to useless, at least if we don't significantly rework how it's computed.

**Comment by ClearlyClaire** [comment]
> Yeah, I think we'd either want to change it to a user search or just a list of all users with a role with permissions

You probably want to be able to investigate former moderators as well

**Comment by ThisIsMissEm** [comment]
So probably a generic user search. n.b., if the former mod had deleted their account, I don't think we've enough data preservation to make this work (see: #31608 )

**Comment by iMagesh** [comment]
Can i work on this? is this still required? @ThisIsMissEm @renchap

**Comment by ThisIsMissEm** [comment]
@iMagesh one potential hold up for this is currently the Audit log only contains Accounts which are moderators or administrators, not regular uses.

As per Claire's comment: https://github.com/mastodon/mastodon/issues/31461#issuecomment-2311881226

**Comment by ThisIsMissEm** [comment]
@iMagesh so if the scope of work includes reworking `/admin/action_logs` to have a user search filter, instead of a dropdown, then that might remove that roadblock, alternatively, we could split that task out, but it'd be a prerequisite for doing this issue.

**Comment by iMagesh** [comment]
Got it @ThisIsMissEm. Thanks for the clarification. 

We can remove the user filter dropdown and make it a generic user search where people must type and select the user who could be an admin, moderator or regular user. We should also start logging user appeals, which can then be filtered eg: "$username has appealed the [account warning]" as per your description. 

We don't have to split that task; I can complete it along with this one as a prerequisite. I will send a PR soon for you to verify.

**Comment by iMagesh** [comment]
> 
> You could look at how I did similar here, but in this I had to handle both local and remote users; in audit log, you only need to handle local users, iirc: [#31449](https://github.com/mastodon/mastodon/pull/31449)

Do you want me to create a search interface like that, with a search button and a static input field for username?

I was thinking maybe we could create an input field that suggests a list of usernames (via a dynamic dropdown) as we type, and then on select, we could list all the logs of the selected user. The user could be anyone here(admin, moderator, generic user). Is that okay? 

Do we have any such smart input field with suggestions used anywhere in this app? I could maybe reuse the component if it exists. 

What do you suggest?

**Comment by ThisIsMissEm** [comment]
@iMagesh on the admin/mod side, we don't really have any component like that, we tend to rely more on server-side filtering — it wouldn't be bad to enrich (progressively enhance) basic text fields with search capabilities, but we're not in the context of the react application, so stuff has to be done differently

**Comment by iMagesh** [comment]
@ThisIsMissEm Yes, we don't have react on the admin side, but can we have a single text field and use some JavaScript to send an API call to fetch users when someone types a minimum set of chars? is that allowed? 

If that's not okay, I can add a simple text field and search button just like you did with reports (#31449), which requires the users to type the correct username in full.

**Comment by ThisIsMissEm** [comment]
@iMagesh you could do that, but then you need to make sure that API is secured, and authenticated correctly, etc.

So for now just doing a text form is probably best.

**Comment by ThisIsMissEm** [comment]
Here, how about we move that to a separate task: https://github.com/mastodon/mastodon/issues/32739

**Comment by iMagesh** [comment]
Thanks @ThisIsMissEm. For now, I will work on creating a simple text form, as seen on the 'reports' page.
